src/main/java/uk/gov/pay/products/persistence/dao/ProductMetadataDao.java

```diff
@@ -33,4 +34,15 @@ public ProductMetadataDao(Provider<EntityManager> entityManager) {
                 .setParameter("productExternalId", productExternalId)
                 .getResultList();
     }
+
+    public Optional<ProductMetadataEntity> findByProductsExternalIdAndKey(String productExternalId, String key) {
+        String query = "SELECT metadata FROM ProductMetadataEntity metadata, ProductEntity product " +
+                "WHERE metadata.productEntity = product AND product.externalId = :productExternalId " +
+                "AND lower(metadata.metadataKey) = :key";
```

Comment:
The index on key is a union with `product_id` `(product_id, metadata_key)` -- I think this should be able to use it but just making sure